dropgz/pkg/embed/payload.go

```diff
@@ -86,6 +86,19 @@ func deploy(src, dest string) error {
 		return err
 	}
 	defer rc.Close()
+
+	if _, err := os.Stat(dest); err == nil || !os.IsNotExist(err) {
+		old_file_dest := dest + ".old"
+		os.RemoveAll(old_file_dest)
+		if err = os.Rename(dest, old_file_dest); err != nil {
+			return errors.Wrapf(err, "not able to rename the %s file to %s old", dest, old_file_dest)
+		}
+	}
+
+	if err != nil {
```

Comment:
the err assignment in line 90 (`if _, err := os.Stat(dest); ...`) creates a local `err` variable in a new scope which shadows the previous err declaration. so this `err != nil` check is actually checking if the err assignment in line 84 is nil, which we did in line 85 already. 
is the err shadowing intentional? if so, which err are you intending to check here?

Reply:
I wanted to check the err is assigned in line 90, incase the err is different from file not present. The err shadow was not intentional. So to actually catch the err of line 90 I can add an else condition to for the line 90 ?